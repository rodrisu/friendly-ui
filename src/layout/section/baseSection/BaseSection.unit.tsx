import React from 'react'
import renderer from 'react-test-renderer'

import { BaseSection, SectionContentSize } from './index'

describe('BaseSection', () => {
  it('should render default basic section', () => {
    const section = <BaseSection>default section</BaseSection>
    const renderedSection = renderer.create(section).toJSON()
    expect(renderedSection).toMatchSnapshot()
  })

  it('should render non-default section', () => {
    const props = {
      tagName: 'article',
      role: 'contentinfo',
      contentSize: SectionContentSize.LARGE,
    }
    const section = <BaseSection {...props}>large section</BaseSection>
    const renderedSection = renderer.create(section).toJSON()
    expect(renderedSection).toMatchSnapshot()
  })
})
