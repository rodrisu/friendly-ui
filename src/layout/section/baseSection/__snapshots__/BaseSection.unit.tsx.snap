// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSection should render default basic section 1`] = `
.c0 .section-content {
  padding-left: 24px;
  padding-right: 24px;
}

.c0 .section-content--noHorizontalSpacing {
  padding-left: 0;
  padding-right: 0;
}

@media (min-width:800px) {
  .c0 .section-content {
    margin-left: auto;
    margin-right: auto;
    max-width: 662px;
  }

  .c0 .section-content.section-content--large {
    max-width: 1016px;
  }
}

<div
  className="c0"
  role="presentation"
>
  <div
    className="section-content"
  >
    default section
  </div>
</div>
`;

exports[`BaseSection should render non-default section 1`] = `
.c0 .section-content {
  padding-left: 24px;
  padding-right: 24px;
}

.c0 .section-content--noHorizontalSpacing {
  padding-left: 0;
  padding-right: 0;
}

@media (min-width:800px) {
  .c0 .section-content {
    margin-left: auto;
    margin-right: auto;
    max-width: 662px;
  }

  .c0 .section-content.section-content--large {
    max-width: 1016px;
  }
}

<div
  className="c0"
  role="presentation"
>
  <article
    className="section-content section-content--large"
    role="contentinfo"
  >
    large section
  </article>
</div>
`;
